protocol/x/clob/keeper/process_operations.go

```diff
@@ -694,7 +694,15 @@ func (k Keeper) PersistMatchDeleveragingToState(
 			metrics.GetLabelForBoolValue(metrics.IsLong, position.GetIsLong()),
 			metrics.GetLabelForBoolValue(metrics.DeliverTx, true),
 		)
-		k.subaccountsKeeper.SetNegativeTncSubaccountSeenAtBlock(ctx, lib.MustConvertIntegerToUint32(ctx.BlockHeight()))
+		collateralPoolAddress, err := k.subaccountsKeeper.GetCollateralPoolFromPerpetualId(ctx, perpetualId)
+		if err != nil {
+			return err
+		}
```

Comment:
nit: can maybe just move this to `SetNegativeTncSubaccountSeenAtBlock`

Reply:
up to you though. no strong preference

Reply:
+1 to jays comment, related to my comment [here](https://github.com/dydxprotocol/v4-chain/pull/1208/files#r1534085785), I think i feel a little more strongly than jay haha